### ISSUE TITLE
numix-folders setting changes with upgrade.

### ISSUE THREAD

**Issue by Himanshu-Mishr** [numixproject/numix-folders#101]
Every time I upgrade numix icons via package manager, settings that I have set for numix-folder gets changed to its default values and colors. How to make my settings permanent ?

**Comment by wa4557** [comment]
@Foggalong I thought about this, and I guess there is a way to do this:

1. Save the current configuration (colour, style, custom etc.) in a config file somewhere in `~/.config` (or somewhere else) by writing it in the end of the script.

2. Add another flag to the script that reads this config file and applies these values without asking the user on the command line

3. On numix-base and circle there is somewhere a postinstall method (that's where the `gtk-update-icon-cache` is done). Add the numix-folder script (using for example `which numix-folders`) with this new flag.

All that's necessary to do this is that `numix-folders` is somewhere in `$PATH` (doesn't matter where exactly)

**Comment by dirtydancing** [comment]
While this issue is a duplicate of https://github.com/numixproject/numix-folders/issues/45, https://github.com/numixproject/numix-folders/issues/47#issuecomment-113933833, https://github.com/numixproject/numix-folders/issues/64 and https://github.com/numixproject/numix-folders/issues/99, we might just address this here. It is apparent that there is a demand for this kind of feature.

I am still not sure whether the method that @wa4557 proposed will work on all distros, or whether this would be specific to updating e.g. via PPA/AUR/copr. How about cloning from GitHub?

An alternative way to address this might be to still have to run the numix-folders script, but to have the script itself provide an option for the last setting, e.g. by entering "lastrun" for the style and be done.

An additional advantage of such an approach would be that Numix base and Numix-Circle stay "independent" of numix-folders in the sense that Numix base and Numix-Circle would not have to include a numix-folders postinstall.

A disadvantage of such an approach would be that the numix-folders script would have to be triggered manually every time Numix base or Numix-Circle updates e.g. via PPA, which can mean on a daily basis.

**Comment by bil-elmoussaoui** [comment]
@dirtydancing @wa4557 i think the best way is to proive a GUI for numix-folders so the user can save the settings in the configuration file

**Comment by idmresettrial** [comment]
@bil-elmoussaoui @wa4557 @Himanshu-Mishr :

**Comment by Foggalong** [comment]
The problem with that script is that it's Ubuntu dependent, so we can't implement that as the solution. As others have mentioned it might be a good idea to implement the save feature along with the GUI as both will require a bit more substance in terms of a proper install method

**Comment by idmresettrial** [comment]
@Foggalong 
Ok, I just let them know about a solution on Ubuntu.
I still wait for an official one from Numix.
^^

**Comment by bil-elmoussaoui** [comment]
I will do that using Python if everyone is ok with that. If anyone have an idea of a possible features or a design of the GUI please share it here.

**Comment by wa4557** [comment]
I guess python is not really necessary here (since the whole script is written in bash), a simple zenity GUI would be sufficient. I'd do it but unfotunately I'm extremely busy at the moment :(

**Comment by bil-elmoussaoui** [comment]
I have no idea how zenity woks, but i will give it a try this weekend.

**Comment by Himanshu-Mishr** [comment]
As a user, I would say saving user's settings somewhere ( in ~/.config ) and using it while upgrading, would be better idea. If config file not present then you can go with default settings. Keep it simple. 
Btw thank you for wonderful icons.

**Comment by Foggalong** [comment]
Yeah, decided I'm gonna keep this open. Enough people have requested it now that it's clearly something people want :P

**Comment by Foggalong** [comment]
CC: @PaoloRotolo

**Comment by xHN35RQ** [comment]
A config file in `~/.config` seems the best way to handle this, and use each distro's post installation routines to run a script that checks for existing config data, and applies the users preferences. This should allow for distro independence.

**Comment by wa4557** [comment]
@xHN35RQ there exists a config file already and running the script with `numix-folders -p` restores the previous settings. 
@Foggalong 
Adding this: `command -v numix-folders -p >/dev/null 2>&1` to the postinstall routine (the same place where the gtk-update-cache is done) should fix the issue for all non git-clone users, and doesn't do any harm if numix-folders is not installed. I really think this should be added

**Comment by Foggalong** [comment]
@PaoloRotolo ^

**Comment by dirtydancing** [comment]
Should this then not be added to Numix-Circle as well? Numix base encompasses the folder icons, but Numix-Circle encompasses the file manager icon which also changes according to the folder style.

**Comment by wa4557** [comment]
True.

**Comment by Foggalong** [comment]
:+1:

**Comment by Foggalong** [comment]
As a note, people using git based installations don't need to worry about this problem anyway since git updates don't cause this problem

**Comment by xHN35RQ** [comment]
> there exists a config file already and running the script with numix-folders -p restores the previous settings. 

@wa4557 Thanks for that info, but I don't want to run this script after every apt-get update. I just want my existing settings to stay the same. :smile:

**Comment by dirtydancing** [comment]
@Foggalong I am not sure what you mean with "git based installations don't need to worry about this problem".

E.g. with Arch/GNOME, when I run `yaourt -Syua --devel --noconfirm` in order to update i.a. all git-based AUR packages (when only the git version has updated without the AUR package itself having been updated), the folder icons are reset to the default as well. (for the Numix base and Numix-Circle AUR packages https://aur.archlinux.org/packages/numix-icon-theme-git/ and https://aur.archlinux.org/packages/numix-circle-icon-theme-git/).

This seems to imply that also the AUR packages would need to include some kind of postinstall routine.

**Comment by Foggalong** [comment]
woah really? I use raw git for managing my Numix install and it doesn't reset so I assumed AUR would be the same. Unfortunately we don't administrate the AUR packages so it's up to the maintainer there to add it. They do occasionally popup on the GitHub though so they might see this :)

**Comment by dirtydancing** [comment]
Yeah, I am not really sure at this point if this is about the AUR package itself, because with git-based AUR packages, they often do not update with versionless "rolling" git projects, but you can still update the git version with the `--devel` flag. To me, this entire situation is not that big of a deal because I can simply run numix-folders (which has saved the previous configuration), and I am done. Anyways, I would agree that for Arch, this is something that the AUR package maintainers would need to address.

**Comment by wa4557** [comment]
But still, adding `$(command -v numix-folders) -p >/dev/null 2>&1`

to base and circle postinstall is a good idea, regardless of the situation for arch packages

**Comment by Foggalong** [comment]
Oh yeah, we'll definitely add it for the PPA :)

**Comment by grenzor** [comment]
+1 folders settings resets to default upon update. any new status on this?

**Comment by Foggalong** [comment]
No update as of yet. @PaoloRotolo will probably announce it here when done :)

**Comment by xHN35RQ** [comment]
Looking forward to this. thanks!

**Comment by morph027** [comment]
Solved it like this:

### ```/usr/local/bin/trigger-numix-folders-blue.sh```

```bash
#!/bin/bash

grep -q '/numix-icon-theme' || exit 0
touch /tmp/trigger-numix-folders-update
```

### ```/usr/local/bin/numix-folders-blue.sh```

```bash
#!/bin/bash

if [ -f /tmp/trigger-numix-folders-update ]; then
echo "restoring blue Numix icons (style 5)"
numix-folders --cli << EOF
5
blue
EOF
rm -f /tmp/trigger-numix-folders-update
fi
```

### ```/etc/apt/apt.conf.d/99-numix-folders-blue```

```
DPkg::Pre-Install-Pkgs { "/usr/local/bin/trigger-numix-folders-blue.sh"; }
DPkg::Post-Invoke { "/usr/local/bin/numix-folders-blue.sh"; }
```

**Comment by morph027** [comment]
https://gitlab.com/snippets/1539528

**Comment by TiZ-EX1** [comment]
All this talk about package manager hooks and more root entrenchment, when all of this could just be solved by creating a theme in `$HOME/.icons` with nothing but the customized folders, that inherits Numix. You could call it Numix-Folders. Then you would update Numix derivatives like Circle such that their inherits line inherits Numix-Folders and then Numix.

Is there a reason this method has been so aggressively avoided? Like, not even mentioned by anyone?

**Comment by bil-elmoussaoui** [comment]
@TiZ-EX1 I'm already working on a new version of Numix Folders that doesn't require root permissions. As with the modifications done to /usr/share we can't really have an official package of it on any distro, netiher a flatpak/snap package. 
i can't tell you when it's going to be ready, as i'm so busy with finals right now but should happen this summer !

**Comment by TiZ-EX1** [comment]
A new version, huh? Well, this current one is written in bash, which I am fairly good at (Don't know if that's a good thing). I'll try to grok it and see what it would take to make it just write the customized icons to `$HOME/.icons`.

**Comment by bil-elmoussaoui** [comment]
It shouldn't be too hard :+1: The new version won't include only this modification, the way the ui was created at first isn't perfect; the code is a little mess and should be cleaned a bit :)

**Comment by TiZ-EX1** [comment]
I just realized that this script modifies Numix-Circle and Numix-Square as well, which makes this a little less straightforward. What would be nice is if user overrides for specific icons could be placed into user directories. I know that userspace apps place icons into "$HOME/.local/share/icons/hicolor", and expand/override parts of the fallback theme that way. If I try to override Numix in that manner (test case is replacing the current oversized wired network icon with the old one), it doesn't seem to work.

**Comment by Foggalong** [comment]
@TiZ-EX1 Yeah, that's why we'd not just done that :P Inheritance trees are a minefield
